core/templates/dev/head/pages/suggestion_editor/ShowSuggestionModalForCreatorViewService.js

```diff
@@ -0,0 +1,133 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service to display suggestion modal in creator view.
+ */
+
+oppia.factory('ShowSuggestionModalForCreatorViewService', [
+  '$uibModal', '$rootScope',
+  'UrlInterpolationService', '$http',
+  function($uibModal, rootScope,
+      UrlInterpolationService, $http) {
+    var _templateUrl = UrlInterpolationService.getDirectiveTemplateUrl(
+      '/pages/suggestion_editor/' +
+      'creator_view_suggestion_modal_directive.html'
+    );
+
+    var _showEditStateContentSuggestionModal = function(
+        activeThread, suggestionsToReviewList, clearActiveThread,
+        canReviewActiveThread) {
+      $uibModal.open({
+        templateUrl: _templateUrl,
+        backdrop: true,
+        size: 'lg',
+        resolve: {
+          suggestionIsHandled: function() {
+            return activeThread.isSuggestionHandled();
+          },
+          suggestionStatus: function() {
+            return activeThread.getSuggestionStatus();
+          },
+          description: function() {
+            return activeThread.description;
+          },
+          oldContent: function() {
+            return activeThread.suggestion.oldValue;
+          },
+          newContent: function() {
+            return activeThread.suggestion.newValue;
+          },
+          canReviewActiveThread: function() {
+            return canReviewActiveThread;
+          },
+          stateName: function() {
+            return activeThread.suggestion.stateName;
```

Comment:
I think this value would be null always (could you check?). If I remember correctly, this field was removed. If so, we can delete this function (and other places where state_name is used).

Reply:
The value is the state name for which the suggestion is provided.

Reply:
While manually testing, did you run into this field being non-null at any point? (IIRC, this was removed from the backend, so I don't think this data would exist.)

Reply:
Just this one question

Reply:
Ah sorry @nithusha21, I missed this one. I actually just tested by adding console.log(activeThread.suggestion.stateName) after starting a local server and that produced a valid statename for which the suggestion was made.

Is there any other path which I should follow for testing this?

Reply:
Oh that's interesting. I think it is fine as is. We can take a look at this independently.

Reply:
Ok, Thanks!